agot-bg-game-server/src/common/ingame-game-state/action-game-state/resolve-march-order-game-state/combat-game-state/choose-house-card-game-state/ChooseHouseCardGameState.ts

```diff
@@ -119,6 +98,47 @@ export default class ChooseHouseCardGameState extends GameState<CombatGameState>
         };
     }
 
+    private checkAndProceedEndOfChooseHouseCardGameState(): void {
+        if (this.houseCards.size == 2) {
+            this.houseCards.forEach((houseCard, house) => this.combatGameState.houseCombatDatas.get(house).houseCard = houseCard);
+
+            // "this.combatGameState.attackingHouseCombatData.houseCard" and
+            // "this.combatGameState.defendingHouseCombatData.houseCard" will always be non-null
+            // since they have just been set before, thus the two "ts-ignore". They could be later set to null
+            // because of Tyrion Lannister, for example.
+            this.ingameGameState.log({
+                type: "combat-house-card-chosen",
+                houseCards: [
+                    // @ts-ignore
+                    [this.combatGameState.attacker.id, this.combatGameState.attackingHouseCombatData.houseCard.id],
+                    // @ts-ignore
+                    [this.combatGameState.defender.id, this.combatGameState.defendingHouseCombatData.houseCard.id]
+                ]
+            });
+
+            this.entireGame.broadcastToClients({
+                type: "change-combat-house-card",
+                // Same here, the houseCards will always be non-null
+                // @ts-ignore
+                houseCardIds: this.combatGameState.houseCombatDatas.map((h, hcd) => [h.id, hcd.houseCard.id])
+            });
+
+            this.combatGameState.onChooseHouseCardGameStateEnd();
+        }
+    }
+
+    private automaticallyChooseLastHouseCard(house: House): void {
+        const choosableCards = this.getChoosableCards(house);
+        if (choosableCards.length == 1) {
+            this.houseCards.set(house, choosableCards[0]);
+
+            this.entireGame.broadcastToClients({
```

Comment:
This is not needed as it will be done during the `firstStart` execution. The client will receive after a serialized version of the GameState with the chosen house card, thus not requiring a packet to be sent here.